### PR TITLE
UCS/SYS: Implement khash for socket address

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

This PR adds common khash function for socket address

## Why ?

Will be used by UCT/TCP transport

## How ?

Implemented Jenkin's hash function that calculates hash code for sockaddr

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9947/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9948/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9950/ for details (Mellanox internal link).

File: src/ucs/sys/sock.c [hunk 1]
```diff
@@ -342,3 +342,104 @@ int ucs_sockaddr_is_equal(const struct sockaddr *sa1,
...
+
+static void ucs_sockaddr_khash_jenkin_mix(uint32_t *hj_i,
+                                          uint32_t *hj_j,
+                                          uint32_t *hj_val)
+{
+    uint32_t a = *hj_i;
+    uint32_t b = *hj_j;
+    uint32_t c = *hj_val;
+    
+    a -= b; a -= c; a ^= (c >> 13);
```
Comment (yosefe) [inline src/ucs/sys/sock.c]: can we reuse ucs_crc16()?

Comment (dmitrygladkov) [reply src/ucs/sys/sock.c]: yes, it makes sense to me. I don't know about this. Closing this PR

**Comment by dmitrygladkov** [comment]
will use `ucs_crc16()` for TCP hash

**Pull request closed without merging**